elastic/handlers.go

```diff
@@ -3,25 +3,33 @@ package elastic
 import (
 	"encoding/json"
 	"fmt"
-	"io/ioutil"
 	"net/http"
 )
 
 func translateHandler(w http.ResponseWriter, r *http.Request) {
-	b, err := ioutil.ReadAll(r.Body)
 	defer r.Body.Close()
-	if err != nil {
+
+	var t Term
+	if err := json.NewDecoder(r.Body).Decode(&t); err != nil {
+		w.WriteHeader(http.StatusUnprocessableEntity)
+
+		json.NewEncoder(w).Encode(map[string]interface{}{
+			"error": err,
+		})
 		return
 	}
-	var t Term
-	json.Unmarshal(b, &t)
 	e, err := Translate(t)
 	if err != nil {
-		http.Error(w, fmt.Sprintf("Query translation returned with error: %v", err), 422)
+		http.Error(w, fmt.Sprintf("Query translation returned with error: %v", err), http.StatusUnprocessableEntity)
```

Comment:
Given that the both other errors are communicated as JSON, this should be define this way as well. A helper function would be good.

```go
func writeJSON(w http.ResponseWriter, status int, data interface{}) {
    if err := json.NewEncoder(w).Encode(data); err != nil {
        w.WriteHeader(http.StatusInternalServerError)
        return
    }
    w.Header().Set("content-type", "application/json")
    w.WriteHeader(status)
}
```